db_stress_tool/expected_state.cc

```diff
@@ -79,6 +79,8 @@ std::vector<PendingExpectedValue> ExpectedState::PrepareDeleteRange(
         PrepareDelete(cf, key, &prepared);
     if (prepared) {
       pending_expected_values.push_back(pending_expected_value);
+    } else {
+      pending_expected_value.Rollback();
```

Comment:
This makes me re-think if reseting the state in destructor instead of explicitly calling Rollback() is actually better than https://github.com/facebook/rocksdb/pull/12123. So if not "committed" in pending_expected_values, it's rollback. Would this be a simpler implementation?

Reply:
This is kind of similar to `Status::PermitUncheckedError` where it serves as a workaround API to mark a `Status` checked without caring about its status. 

We want to achieve similar thing here, we created a `PendingExpectedState` but we don't care if it's committed or rolled back, we just want to make sure its pending state is no longer pending. Because we are not passing this `PendingExpectedState` any further for others to close the pending state.

I used `Rollback` as a workaround, a better approach would probably be to add a separate and dedicate API, something like `PendingExpectedValue::PermitUnclosedPendingState`.

The approach you mentioned could work too, we need to handle similar issues like the code in the constructor and assignment operator is doing. When a `PendingExpectedValue` is being destructed just because it's copied over or moved to another `PendingExpectedValue`, we need to have some special handling. I think the `PermitUnclosedPendingState` approach is easier to understand, especially because it's analogous to how `Status` is handling the checked flag.

Reply:
PermitUnclosedPendingState() is good with me to be used here instead of rollback().